compiler/parser/statement_parsing.go

```diff
@@ -51,6 +51,11 @@ func (p *Parser) parseDefMethodStatement() *ast.DefStatement {
 
 	p.nextToken()
 
+	if !p.curTokenIs(token.Ident) && !(p.peekTokenIs(token.Dot) && (p.curTokenIs(token.InstanceVariable) || p.curTokenIs(token.Constant) || p.curTokenIs(token.Self))) {
```

Comment:
I think you should refactor this to like `!p.IsArgumentToken()`

Reply:
thanks, I will refactor it